OpenSourcePolitics/decidim-app
# feat: Add decidim-cleaner

#### :tophat: Description

Addition of the module decidim cleaner to the decidim-app for the 2.5 release

#### :pushpin: Related Issues
*Link your PR to an issue*
- [Release 2.5 - Ajout module decidim-cleaner](https://opensourcepolitics.odoo.com/web?db=opensourcepolitics&signup_email=guillaume%40opensourcepolitics.eu&token=ixqNZ9S5lGRCj9ADpsQq#id=2871&cids=1&menu_id=325&action=471&model=project.task&view_type=form)

#### Testing
## 🚨 THIS IS A TRICKY TEST, PLEASE IF YOU ARE HAVING ANY ISSUES DON'T MIND CONTACTING ME TO MAKE IT IN PEER TESTING

### Setup 

* Log in as admin
* Access Backoffice
* Go to organization settings
* Access to the "Data Cleaner"
* Enable both of the tests and put "1" or "2" whatever we'll actually test for a longer duration than that.

### Technical Part

- Go to your IDE (RubyMine would be the greatest one)
- Click on the DB Interface
- Click on the "+"
- Select Data Source PostgreSQL
- When the page opens access Schemas
- Click All databases and all schemas and validate
- Click OK and Select on your interface the thing you just created (must be a name like postgres@localhost)
- Search for decidim_app
- Click on "public"
- Click on "tables"
- Search for "decidim-users" and double click on it
- Target a user (preferably the user with the id "2" which may correspond to "user@example.org"

## Test the deletion of a user

- Log in on the instance as user@example.org
- Go back to the DB and check if the columns `current_sign_in_at` and `last_signin_at` were updated
- If they were, please update it and go back 2 years earlier.
- Now run the command `bundle exec rake decidim_cleaner:clean_inactive_users`
- Check that for the same user the column `warning_date` has been updated
- If it was, please update it and go back 1 year earlier.
- Rerun the command `bundle exec rake decidim_cleaner:clean_inactive_users`
- Make sure the column `deleted_at` has been updated and that the reason has been greatly fulfilled in the column next to it `delete_reason`


If everything worked correctly that means that the decidim_cleaner is working pretty well.

#### Tasks
- [x] Add the module cleaner to the decidim-app
- [x] Update the .env-example with the new env variables
- [x] Add new migration from decidim-cleaner module
- [x] Update the sidekiq queues to add new ones
- [x] Add a system test related to the back-office enabling of the cleaner

### :camera: Screenshots
*Please add screenshots of the changes you're proposing if related to the UI*
